src/test/features/offer/routes/your-offer.ts

```diff
@@ -0,0 +1,130 @@
+
+  describe('on GET', () => {
+    checkAuthorizationGuards(app, 'get', offerPage)
+
+    context('when user authorised', () => {
+      beforeEach(() => {
+        idamServiceMock.resolveRetrieveUserFor('1', 'cmc-private-beta', 'defendant')
+      })
+
+      it('should return 500 and render error page when cannot retrieve claims', async () => {
+        claimStoreServiceMock.rejectRetrieveClaimByExternalId('HTTP error')
+
+        await request(app)
+          .get(offerPage)
+          .set('Cookie', `${cookieName}=ABC`)
+          .expect(res => expect(res).to.be.serverError.withText('Error'))
+      })
+
+      it('should render page when everything is fine', async () => {
+        claimStoreServiceMock.resolveRetrieveClaimByExternalId()
+        await request(app)
+          .get(offerPage)
+          .set('Cookie', `${cookieName}=ABC`)
+          .expect(res => expect(res).to.be.successful.withText('Your offer'))
+      })
+    })
+
+    describe('on POST', () => {
+      checkAuthorizationGuards(app, 'post', offerPage)
+
+      context('when user authorised', () => {
+        beforeEach(() => {
+          idamServiceMock.resolveRetrieveUserFor('1', 'cmc-private-beta', 'defendant')
+        })
+
+        context('when middleware failure', () => {
+          it('should return 500 when cannot retrieve claim by external id', async () => {
+            claimStoreServiceMock.rejectRetrieveClaimByExternalId('HTTP error')
+
+            await request(app)
+              .post(offerPage)
+              .set('Cookie', `${cookieName}=ABC`)
+              .send(validFormData)
+              .expect(res => expect(res).to.be.serverError.withText('Error'))
+          })
+        })
+
+        context('when form is valid', async () => {
+          it('should redirect to offer confirmation page', async () => {
+            claimStoreServiceMock.resolveRetrieveClaimByExternalId()
+            claimStoreServiceMock.resolveSaveOffer()
+            const futureDate = moment().add(4, 'month')
+            const formData = {
+              offerText: 'Offer Text',
+              completionDate: new LocalDate(futureDate.year(), futureDate.month(), futureDate.days())
+            }
+            await request(app)
+              .post(offerPage)
+              .set('Cookie', `${cookieName}=ABC`)
+              .send(formData)
+              .expect(res => expect(res).to.be.redirect.toLocation(confirmationPage))
+          })
+
+          it('should return 500 and render error page when cannot save offer', async () => {
+            claimStoreServiceMock.resolveRetrieveClaimByExternalId()
+            claimStoreServiceMock.rejectSaveOfferForDefendant()
+
+            await request(app)
+              .post(offerPage)
+              .set('Cookie', `${cookieName}=ABC`)
+              .send(validFormData)
+              .expect(res => expect(res).to.be.serverError.withText('Error'))
+          })
+        })
+
+        context('when form is invalid', async () => {
```

Comment:
Since the test label for next test is "when provided date is in past", can you please update this one to be "when offer text is not provided"?

Reply:
test checks for invalid form that includes offer text and completionDate

Reply:
If the test's intent is to just check if validation triggers and displays a proper page then either this test case or the one below is redundant on the routes level. Specific validations are already tested on the form object unit test level.